Modules/Quotas/src/Quotas.Application/Metrics/Commands/RecalculateMetricStatuses/Handler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Backbone.Modules.Quotas.Application.Extensions;
+using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain;
+using MediatR;
+
+namespace Backbone.Modules.Quotas.Application.Metrics.Commands.RecalculateMetricStatuses;
+public class Handler : IRequestHandler<RecalculateMetricStatusesCommand>
+{
+    private readonly IMetricCalculatorFactory _metricCalculatorFactory;
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly IQuotasRepository _quotasRepository;
+
+    public Handler(IMetricCalculatorFactory metricCalculatorFactory, IIdentitiesRepository identitiesRepository, IQuotasRepository quotasRepository)
+    {
+        _metricCalculatorFactory = metricCalculatorFactory;
+        _identitiesRepository = identitiesRepository;
+        _quotasRepository = quotasRepository;
+    }
+
+    public async Task Handle(RecalculateMetricStatusesCommand command, CancellationToken cancellationToken)
+    {
+        foreach (var identityId in command.Identities)
+        {
+            var identity = await _identitiesRepository.FindById(identityId, cancellationToken); 
+            await identity.UpdateMetrics(command.Metrics, _metricCalculatorFactory, _quotasRepository, cancellationToken);
```

Comment:
This is very inefficient. You should read all Identities first, then update all of them one by one, and finally save all of them at once.

I think you should verify this with a test so that nobody has the idea to change this in the future (use FakeItEasy-mocks to verify that the methods are called exactly X times)